File: purchasing_controller.py
from javafx.event import ActionEvent
from javafx.fxml import FXML
from javafx.scene.control import Label
from javafx.scene.paint import Color

class PurchasingController:
    @FXML
    private Label MQ4CTally
    @FXML
    private Label MQ4CGCSTally
    @FXML
    private Label MQ4CSpend
    @FXML
    private Label MQ4COAndS
    @FXML
    private Label MQ4CPersonnel
    @FXML
    private Label MQ9BTally
    @FXML
    private Label MQ9BGCSTally
    @FXML
    private Label MQ9BSpend
    @FXML
    private Label MQ9BOAndS
    @FXML
    private Label MQ9BPersonnel
    @FXML
    private Label P1Tally
    @FXML
    private Label P1Spend
    @FXML
    private Label P1OAndS
    @FXML
    private Label P1Personnel
    @FXML
    private Label integratorTally
    @FXML
    private Label integratorGCSTally
    @FXML
    private Label integratorSpend
    @FXML
    private Label integratorOAndS
    @FXML
    private Label integratorPersonnel
    @FXML
    private Label RQ58Tally
    @FXML
    private Label RQ58GCSTally
    @FXML
    private Label RQ58Spend
    @FXML
    private Label RQ58OAndS
    @FXML
    private Label RQ58Personnel
    @FXML
    private Label thunderheadTally
    @FXML
    private Label thunderheadGCSTally
    @FXML
    private Label thunderheadSpend
    @FXML
    private Label thunderheadOAndS
    @FXML
    private Label thunderheadPersonnel
    @FXML
    private Label UAVPurchased
    @FXML
    private Label GCSPurchased
    @FXML
    private Label totalSpend
    @FXML
    private Label totalOAndS
    @FXML
    private Label totalPersonnel
    @FXML
    private Label budget

    def __init__(self):
        pass

    @FXML
    private def onMQ4CIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.MQ4CTally)
        if curr == 0:
            self.MQ4CTally.setText("4")
            if MainController.getInt(self.MQ4CGCSTally) < 2:
                self.MQ4CGCSTally.setText("2")
        else:
            self.MQ4CTally.setText(str(curr + 1))

        self.onTickerClick()

    @FXML
    private def onMQ4CDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.MQ4CTally)
        if curr != 0:
            if curr == 4:
                self.MQ4CTally.setText("0")
                self.MQ4CGCSTally.setText("0")
            else:
                self.MQ4CTally.setText(str(curr - 1))

            self.onTickerClick()

    @FXML
    private def onMQ4CGCSIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.MQ4CGCSTally)
        if MainController.getInt(self.MQ4CTally) > 0:
            self.MQ4CGCSTally.setText(str(curr + 1))
            self.onTickerClick()

    @FXML
    private def onMQ4CGCSDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.MQ4CGCSTally)
        if curr != 0 and (MainController.getInt(self.MQ4CTally) <= 0 or curr > 2):
            self.MQ4CGCSTally.setText(str(curr - 1))
            self.onTickerClick()

    @FXML
    private def onMQ9BIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.MQ9BTally)
        if curr == 0:
            self.MQ9BTally.setText("4")
            if MainController.getInt(self.MQ9BGCSTally) < 2:
                self.MQ9BGCSTally.setText("2")
        else:
            self.MQ9BTally.setText(str(curr + 1))

        self.onTickerClick()

    @FXML
    private def onMQ9BDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.MQ9BTally)
        if curr != 0:
            if curr == 4:
                self.MQ9BTally.setText("0")
                self.MQ9BGCSTally.setText("0")
            else:
                self.MQ9BTally.setText(str(curr - 1))

            self.onTickerClick()

    @FXML
    private def onMQ9BGCSIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.MQ9BGCSTally)
        if MainController.getInt(self.MQ9BTally) > 0:
            self.MQ9BGCSTally.setText(str(curr + 1))
            self.onTickerClick()

    @FXML
    private def onMQ9BGCSDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.MQ9BGCSTally)
        if curr != 0 and (MainController.getInt(self.MQ9BTally) <= 0 or curr > 2):
            self.MQ9BGCSTally.setText(str(curr - 1))
            self.onTickerClick()

    @FXML
    private def onIntegratorIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.integratorTally)
        if curr == 0:
            self.integratorTally.setText("4")
            if MainController.getInt(self.integratorGCSTally) < 2:
                self.integratorGCSTally.setText("2")
        else:
            self.integratorTally.setText(str(curr + 1))

        self.onTickerClick()

    @FXML
    private def onIntegratorDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.integratorTally)
        if curr != 0:
            if curr == 4:
                self.integratorTally.setText("0")
                self.integratorGCSTally.setText("0")
            else:
                self.integratorTally.setText(str(curr - 1))

            self.onTickerClick()

    @FXML
    private def onIntegratorGCSIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.integratorGCSTally)
        if MainController.getInt(self.integratorTally) > 0:
            self.integratorGCSTally.setText(str(curr + 1))
            self.onTickerClick()

    @FXML
    private def onIntegratorGCSDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.integratorGCSTally)
        if curr != 0 and (MainController.getInt(self.integratorTally) <= 0 or curr > 2):
            self.integratorGCSTally.setText(str(curr - 1))
            self.onTickerClick()

    @FXML
    private def onP1Increment(self, e: ActionEvent):
        curr = MainController.getInt(self.P1Tally)
        self.P1Tally.setText(str(curr + 1))
        self.onTickerClick()

    @FXML
    private def onP1Decrement(self, e: ActionEvent):
        curr = MainController.getInt(self.P1Tally)
        if curr != 0:
            self.P1Tally.setText(str(curr - 1))
            self.onTickerClick()

    @FXML
    private def onRQ58Increment(self, e: ActionEvent):
        curr = MainController.getInt(self.RQ58Tally)
        if curr == 0:
            self.RQ58Tally.setText("4")
            if MainController.getInt(self.RQ58GCSTally) < 2:
                self.RQ58GCSTally.setText("2")
        else:
            self.RQ58Tally.setText(str(curr + 1))

        self.onTickerClick()

    @FXML
    private def onRQ58Decrement(self, e: ActionEvent):
        curr = MainController.getInt(self.RQ58Tally)
        if curr != 0:
            if curr == 4:
                self.RQ58Tally.setText("0")
                self.RQ58GCSTally.setText("0")
            else:
                self.RQ58Tally.setText(str(curr - 1))

            self.onTickerClick()

    @FXML
    private def onRQ58GCSIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.RQ58GCSTally)
        if MainController.getInt(self.RQ58Tally) > 0:
            self.RQ58GCSTally.setText(str(curr + 1))
            self.onTickerClick()

    @FXML
    private def onRQ58GCSDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.RQ58GCSTally)
        if curr != 0 and (MainController.getInt(self.RQ58Tally) <= 0 or curr > 2):
            self.RQ58GCSTally.setText(str(curr - 1))
            self.onTickerClick()

    @FXML
    private def onThunderheadIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.thunderheadTally)
        if curr == 0:
            self.thunderheadTally.setText("4")
            if MainController.getInt(self.thunderheadGCSTally) < 2:
                self.thunderheadGCSTally.setText("2")
        else:
            self.thunderheadTally.setText(str(curr + 1))

        self.onTickerClick()

    @FXML
    private def onThunderheadDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.thunderheadTally)
        if curr != 0:
            if curr == 4:
                self.thunderheadTally.setText("0")
                self.thunderheadGCSTally.setText("0")
            else:
                self.thunderheadTally.setText(str(curr - 1))

            self.onTickerClick()

    @FXML
    private def onThunderheadGCSIncrement(self, e: ActionEvent):
        curr = MainController.getInt(self.thunderheadGCSTally)
        if MainController.getInt(self.thunderheadTally) > 0:
            self.thunderheadGCSTally.setText(str(curr + 1))
            self.onTickerClick()

    @FXML
    private def onBudgetDecrement(self, e: ActionEvent):
        curr = PurchasingBackend.cart.initialBudget
        if curr > 100000000:
            PurchasingBackend.cart.initialBudget -= 100000000
            self.onTickerClick()

    @FXML
    private def onBudgetIncrement(self, e: ActionEvent):
        PurchasingBackend.cart.initialBudget += 100000000
        self.onTickerClick()

    @FXML
    private def onThunderheadGCSDecrement(self, e: ActionEvent):
        curr = MainController.getInt(self.thunderheadGCSTally)
        if curr != 0 and (MainController.getInt(self.thunderheadTally) <= 0 or curr > 2):
            self.thunderheadGCSTally.setText(str(curr - 1))
            self.onTickerClick()

    def populate(self):
        self.MQ4CTally.setText(str(PurchasingBackend.cart.MQ4C.UAVPurchased))
        self.MQ9BTally.setText(str(PurchasingBackend.cart.MQ9B.UAVPurchased))
        self.P1Tally.setText(str(PurchasingBackend.cart.P1.UAVPurchased))
        self.integratorTally.setText(str(PurchasingBackend.cart.integrator.UAVPurchased))
        self.RQ58Tally.setText(str(PurchasingBackend.cart.RQ58.UAVPurchased))
        self.thunderheadTally.setText(str(PurchasingBackend.cart.thunderhead.UAVPurchased))
        self.MQ4CGCSTally.setText(str(PurchasingBackend.cart.MQ4C.GCSPurchased))
        self.MQ9BGCSTally.setText(str(PurchasingBackend.cart.MQ9B.GCSPurchased))
        self.integratorGCSTally.setText(str(PurchasingBackend.cart.integrator.GCSPurchased))
        self.RQ58GCSTally.setText(str(PurchasingBackend.cart.RQ58.GCSPurchased))
        self.thunderheadGCSTally.setText(str(PurchasingBackend.cart.thunderhead.GCSPurchased))
        self.MQ4CSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.MQ4C.getTotalCost()))
        self.MQ9BSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.MQ9B.getTotalCost()))
        self.integratorSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.integrator.getTotalCost()))
        self.P1Spend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.P1.getTotalCost()))
        self.RQ58Spend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.RQ58.getTotalCost()))
        self.thunderheadSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.thunderhead.getTotalCost()))
        self.MQ4COAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.MQ4C.getOAndSCost()))
        self.MQ9BOAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.MQ9B.getOAndSCost()))
        self.integratorOAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.integrator.getOAndSCost()))
        self.P1OAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.P1.getOAndSCost()))
        self.RQ58OAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.RQ58.getOAndSCost()))
        self.thunderheadOAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.thunderhead.getOAndSCost()))
        self.MQ4CPersonnel.setText(str(PurchasingBackend.cart.MQ4C.getPersonnelRequirement()))
        self.MQ9BPersonnel.setText(str(PurchasingBackend.cart.MQ9B.getPersonnelRequirement()))
        self.integratorPersonnel.setText(str(PurchasingBackend.cart.integrator.getPersonnelRequirement()))
        self.P1Personnel.setText(str(PurchasingBackend.cart.P1.getPersonnelRequirement()))
        self.RQ58Personnel.setText(str(PurchasingBackend.cart.RQ58.getPersonnelRequirement()))
        self.thunderheadPersonnel.setText(str(PurchasingBackend.cart.thunderhead.getPersonnelRequirement()))
        self.UAVPurchased.setText(str(PurchasingBackend.cart.getTotalUAV()))
        self.GCSPurchased.setText(str(PurchasingBackend.cart.getTotalGCS()))
        self.totalSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.getTotalSpend()))
        self.totalOAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.getOAndSSpend()))
        self.totalPersonnel.setText(str(PurchasingBackend.cart.getPersonnelRequirement()))
        self.budget.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.initialBudget))

    def alert(self):
        if PurchasingBackend.cart.getTotalSpend() > PurchasingBackend.cart.initialBudget:
            self.totalSpend.setTextFill(Color.rgb(161, 0, 0))
        else:
            self.totalSpend.setTextFill(Color.BLACK)

    private def onTickerClick(self):
        PurchasingBackend.cart.MQ4C.UAVPurchased = MainController.getInt(self.MQ4CTally)
        PurchasingBackend.cart.MQ9B.UAVPurchased = MainController.getInt(self.MQ9BTally)
        PurchasingBackend.cart.P1.UAVPurchased = MainController.getInt(self.P1Tally)
        PurchasingBackend.cart.integrator.UAVPurchased = MainController.getInt(self.integratorTally)
        PurchasingBackend.cart.RQ58.UAVPurchased = MainController.getInt(self.RQ58Tally)
        PurchasingBackend.cart.thunderhead.UAVPurchased = MainController.getInt(self.thunderheadTally)
        PurchasingBackend.cart.MQ4C.GCSPurchased = MainController.getInt(self.MQ4CGCSTally)
        PurchasingBackend.cart.MQ9B.GCSPurchased = MainController.getInt(self.MQ9BGCSTally)
        PurchasingBackend.cart.integrator.GCSPurchased = MainController.getInt(self.integratorGCSTally)
        PurchasingBackend.cart.RQ58.GCSPurchased = MainController.getInt(self.RQ58GCSTally)
        PurchasingBackend.cart.thunderhead.GCSPurchased = MainController.getInt(self.thunderheadGCSTally)
        self.populate()
        self.alert()

    def initialize(self):
        self.populate()
        self.alert()

File: PurchasingController.java
package com.example.demo;

import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.paint.Color;

public class PurchasingController {
    @FXML private Label MQ4CTally, MQ4CGCSTally, MQ4CSpend, MQ4COAndS, MQ4CPersonnel;
    @FXML private Label MQ9BTally, MQ9BGCSTally, MQ9BSpend, MQ9BOAndS, MQ9BPersonnel;
    @FXML private Label P1Tally, P1Spend, P1OAndS, P1Personnel;
    @FXML private Label integratorTally, integratorGCSTally, integratorSpend, integratorOAndS, integratorPersonnel;
    @FXML private Label RQ58Tally, RQ58GCSTally, RQ58Spend, RQ58OAndS, RQ58Personnel;
    @FXML private Label thunderheadTally, thunderheadGCSTally, thunderheadSpend, thunderheadOAndS, thunderheadPersonnel;
    @FXML private Label UAVPurchased, GCSPurchased, totalSpend, totalOAndS, totalPersonnel;
    @FXML private Label budget;
    @FXML private void onMQ4CIncrement(ActionEvent e) {
        int curr = MainController.getInt(MQ4CTally);
        if(curr == 0) {
            MQ4CTally.setText("4");
            if(MainController.getInt(MQ4CGCSTally) < 2) {
                MQ4CGCSTally.setText("2");
            }
        }
        else {
            MQ4CTally.setText("" + (curr + 1));
        }
        onTickerClick();
    }
    @FXML private void onMQ4CDecrement(ActionEvent e) {
        int curr = MainController.getInt(MQ4CTally);
        if(curr == 0) {
            return;
        }
        if(curr == 4) {
            MQ4CTally.setText("0");
            MQ4CGCSTally.setText("0");
        }
        else {
            MQ4CTally.setText("" + (curr - 1));
        }
        onTickerClick();
    }
    @FXML private void onMQ4CGCSIncrement(ActionEvent e) {
        int curr = MainController.getInt(MQ4CGCSTally);
        if(MainController.getInt(MQ4CTally) <= 0) {
            return;
        }
        MQ4CGCSTally.setText("" + (curr + 1));
        onTickerClick();
    }
    @FXML private void onMQ4CGCSDecrement(ActionEvent e) {
        int curr = MainController.getInt(MQ4CGCSTally);
        if((curr == 0) || ((MainController.getInt(MQ4CTally) > 0) && (curr <= 2))) {
            return;
        }
        MQ4CGCSTally.setText("" + (curr - 1));
        onTickerClick();
    }
    @FXML private void onMQ9BIncrement(ActionEvent e) {
        int curr = MainController.getInt(MQ9BTally);
        if(curr == 0) {
            MQ9BTally.setText("4");
            if(MainController.getInt(MQ9BGCSTally) < 2) {
                MQ9BGCSTally.setText("2");
            }
        }
        else {
            MQ9BTally.setText("" + (curr + 1));
        }
        onTickerClick();
    }
    @FXML private void onMQ9BDecrement(ActionEvent e) {
        int curr = MainController.getInt(MQ9BTally);
        if(curr == 0) {
            return;
        }
        if(curr == 4) {
            MQ9BTally.setText("0");
            MQ9BGCSTally.setText("0");
        }
        else {
            MQ9BTally.setText("" + (curr - 1));
        }
        onTickerClick();
    }
    @FXML private void onMQ9BGCSIncrement(ActionEvent e) {
        int curr = MainController.getInt(MQ9BGCSTally);
        if(MainController.getInt(MQ9BTally) <= 0) {
            return;
        }
        MQ9BGCSTally.setText("" + (curr + 1));
        onTickerClick();
    }
    @FXML private void onMQ9BGCSDecrement(ActionEvent e) {
        int curr = MainController.getInt(MQ9BGCSTally);
        if((curr == 0) || ((MainController.getInt(MQ9BTally) > 0) && (curr <= 2))) {
            return;
        }
        MQ9BGCSTally.setText("" + (curr - 1));
        onTickerClick();
    }
    @FXML private void onIntegratorIncrement(ActionEvent e) {
        int curr = MainController.getInt(integratorTally);
        if(curr == 0) {
            integratorTally.setText("4");
            if(MainController.getInt(integratorGCSTally) < 2) {
                integratorGCSTally.setText("2");
            }
        }
        else {
            integratorTally.setText("" + (curr + 1));
        }
        onTickerClick();
    }
    @FXML private void onIntegratorDecrement(ActionEvent e) {
        int curr = MainController.getInt(integratorTally);
        if(curr == 0) {
            return;
        }
        if(curr == 4) {
            integratorTally.setText("0");
            integratorGCSTally.setText("0");
        }
        else {
            integratorTally.setText("" + (curr - 1));
        }
        onTickerClick();
    }
    @FXML private void onIntegratorGCSIncrement(ActionEvent e) {
        int curr = MainController.getInt(integratorGCSTally);
        if(MainController.getInt(integratorTally) <= 0) {
            return;
        }
        integratorGCSTally.setText("" + (curr + 1));
        onTickerClick();
    }
    @FXML private void onIntegratorGCSDecrement(ActionEvent e) {
        int curr = MainController.getInt(integratorGCSTally);
        if((curr == 0) || ((MainController.getInt(integratorTally) > 0) && (curr <= 2))) {
            return;
        }
        integratorGCSTally.setText("" + (curr - 1));
        onTickerClick();
    }
    @FXML private void onP1Increment(ActionEvent e) {
        int curr = MainController.getInt(P1Tally);
        P1Tally.setText("" + (curr + 1));
        onTickerClick();
    }
    @FXML private void onP1Decrement(ActionEvent e) {
        int curr = MainController.getInt(P1Tally);
        if(curr == 0) {
            return;
        }
        P1Tally.setText("" + (curr - 1));
        onTickerClick();
    }
    @FXML private void onRQ58Increment(ActionEvent e) {
        int curr = MainController.getInt(RQ58Tally);
        if(curr == 0) {
            RQ58Tally.setText("4");
            if(MainController.getInt(RQ58GCSTally) < 2) {
                RQ58GCSTally.setText("2");
            }
        }
        else {
            RQ58Tally.setText("" + (curr + 1));
        }
        onTickerClick();
    }
    @FXML private void onRQ58Decrement(ActionEvent e) {
        int curr = MainController.getInt(RQ58Tally);
        if(curr == 0) {
            return;
        }
        if(curr == 4) {
            RQ58Tally.setText("0");
            RQ58GCSTally.setText("0");
        }
        else {
            RQ58Tally.setText("" + (curr - 1));
        }
        onTickerClick();
    }
    @FXML private void onRQ58GCSIncrement(ActionEvent e) {
        int curr = MainController.getInt(RQ58GCSTally);
        if(MainController.getInt(RQ58Tally) <= 0) {
            return;
        }
        RQ58GCSTally.setText("" + (curr + 1));
        onTickerClick();
    }
    @FXML private void onRQ58GCSDecrement(ActionEvent e) {
        int curr = MainController.getInt(RQ58GCSTally);
        if((curr == 0) || ((MainController.getInt(RQ58Tally) > 0) && (curr <= 2))) {
            return;
        }
        RQ58GCSTally.setText("" + (curr - 1));
        onTickerClick();
    }
    @FXML private void onThunderheadIncrement(ActionEvent e) {
        int curr = MainController.getInt(thunderheadTally);
        if(curr == 0) {
            thunderheadTally.setText("4");
            if(MainController.getInt(thunderheadGCSTally) < 2) {
                thunderheadGCSTally.setText("2");
            }
        }
        else {
            thunderheadTally.setText("" + (curr + 1));
        }
        onTickerClick();
    }
    @FXML private void onThunderheadDecrement(ActionEvent e) {
        int curr = MainController.getInt(thunderheadTally);
        if(curr == 0) {
            return;
        }
        if(curr == 4) {
            thunderheadTally.setText("0");
            thunderheadGCSTally.setText("0");
        }
        else {
            thunderheadTally.setText("" + (curr - 1));
        }
        onTickerClick();
    }
    @FXML private void onThunderheadGCSIncrement(ActionEvent e) {
        int curr = MainController.getInt(thunderheadGCSTally);
        if(MainController.getInt(thunderheadTally) <= 0) {
            return;
        }
        thunderheadGCSTally.setText("" + (curr + 1));
        onTickerClick();
    }
    @FXML private void onBudgetDecrement(ActionEvent e) {
        long curr = PurchasingBackend.cart.initialBudget;
        if(curr <= 100000000) {
            return;
        }
        PurchasingBackend.cart.initialBudget -= 100000000;
        onTickerClick();
    }
    @FXML private void onBudgetIncrement(ActionEvent e) {
        PurchasingBackend.cart.initialBudget += 100000000;
        onTickerClick();
    }
    @FXML private void onThunderheadGCSDecrement(ActionEvent e) {
        int curr = MainController.getInt(thunderheadGCSTally);
        if((curr == 0) || ((MainController.getInt(thunderheadTally) > 0) && (curr <= 2))) {
            return;
        }
        thunderheadGCSTally.setText("" + (curr - 1));
        onTickerClick();
    }
    public void populate() {
        //Populate UAV tallies
        MQ4CTally.setText("" + PurchasingBackend.cart.MQ4C.UAVPurchased);
        MQ9BTally.setText("" + PurchasingBackend.cart.MQ9B.UAVPurchased);
        P1Tally.setText("" + PurchasingBackend.cart.P1.UAVPurchased);
        integratorTally.setText("" + PurchasingBackend.cart.integrator.UAVPurchased);
        RQ58Tally.setText("" + PurchasingBackend.cart.RQ58.UAVPurchased);
        thunderheadTally.setText("" + PurchasingBackend.cart.thunderhead.UAVPurchased);
        //Populate GCS tallies
        MQ4CGCSTally.setText("" + PurchasingBackend.cart.MQ4C.GCSPurchased);
        MQ9BGCSTally.setText("" + PurchasingBackend.cart.MQ9B.GCSPurchased);
        integratorGCSTally.setText("" + PurchasingBackend.cart.integrator.GCSPurchased);
        RQ58GCSTally.setText("" + PurchasingBackend.cart.RQ58.GCSPurchased);
        thunderheadGCSTally.setText("" + PurchasingBackend.cart.thunderhead.GCSPurchased);
        //Populates procurement prices
        MQ4CSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.MQ4C.getTotalCost()));
        MQ9BSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.MQ9B.getTotalCost()));
        integratorSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.integrator.getTotalCost()));
        P1Spend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.P1.getTotalCost()));
        RQ58Spend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.RQ58.getTotalCost()));
        thunderheadSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.thunderhead.getTotalCost()));
        //Populates O&S prices
        MQ4COAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.MQ4C.getOAndSCost()));
        MQ9BOAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.MQ9B.getOAndSCost()));
        integratorOAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.integrator.getOAndSCost()));
        P1OAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.P1.getOAndSCost()));
        RQ58OAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.RQ58.getOAndSCost()));
        thunderheadOAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.thunderhead.getOAndSCost()));
        //Populates personnel requirement
        MQ4CPersonnel.setText("" + PurchasingBackend.cart.MQ4C.getPersonnelRequirement());
        MQ9BPersonnel.setText("" + PurchasingBackend.cart.MQ9B.getPersonnelRequirement());
        integratorPersonnel.setText("" + PurchasingBackend.cart.integrator.getPersonnelRequirement());
        P1Personnel.setText("" + PurchasingBackend.cart.P1.getPersonnelRequirement());
        RQ58Personnel.setText("" + PurchasingBackend.cart.RQ58.getPersonnelRequirement());
        thunderheadPersonnel.setText("" + PurchasingBackend.cart.thunderhead.getPersonnelRequirement());
        //Populates totals
        UAVPurchased.setText("" + PurchasingBackend.cart.getTotalUAV());
        GCSPurchased.setText("" + PurchasingBackend.cart.getTotalGCS());
        totalSpend.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.getTotalSpend()));
        totalOAndS.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.getOAndSSpend()));
        totalPersonnel.setText("" + PurchasingBackend.cart.getPersonnelRequirement());
        budget.setText(PurchasingBackend.priceFormat(PurchasingBackend.cart.initialBudget));
    }
    public void alert() {
        if((PurchasingBackend.cart.getTotalSpend() > PurchasingBackend.cart.initialBudget)) {
            totalSpend.setTextFill(Color.rgb(161, 0, 0));
        } else {
            totalSpend.setTextFill(Color.BLACK);
        }
    }
    private void onTickerClick() {
        PurchasingBackend.cart.MQ4C.UAVPurchased = MainController.getInt(MQ4CTally);
        PurchasingBackend.cart.MQ9B.UAVPurchased = MainController.getInt(MQ9BTally);
        PurchasingBackend.cart.P1.UAVPurchased = MainController.getInt(P1Tally);
        PurchasingBackend.cart.integrator.UAVPurchased = MainController.getInt(integratorTally);
        PurchasingBackend.cart.RQ58.UAVPurchased = MainController.getInt(RQ58Tally);
        PurchasingBackend.cart.thunderhead.UAVPurchased = MainController.getInt(thunderheadTally);
        PurchasingBackend.cart.MQ4C.GCSPurchased = MainController.getInt(MQ4CGCSTally);
        PurchasingBackend.cart.MQ9B.GCSPurchased = MainController.getInt(MQ9BGCSTally);
        PurchasingBackend.cart.integrator.GCSPurchased = MainController.getInt(integratorGCSTally);
        PurchasingBackend.cart.RQ58.GCSPurchased = MainController.getInt(RQ58GCSTally);
        PurchasingBackend.cart.thunderhead.GCSPurchased = MainController.getInt(thunderheadGCSTally);
        populate();
        alert();
    }
    public void initialize() {
        populate();
        alert();
    }
}
